src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,332 @@
+                    extra = {
+                        "organization_id": project.organization.id,
+                        "project_id": project.id,
+                        "group_data": json.dumps(rows),
+                        "error": e.message,
+                    }
+                    logger.info(
+                        "tasks.backfill_seer_grouping_records.bulk_event_lookup_exception",
+                        extra=extra,
+                    )
+
+        group_data = []
+        stacktrace_strings = []
+        bulk_event_ids = set()
+        invalid_event_ids = set()
+        with sentry_sdk.start_transaction(op="embeddings_grouping.get_latest_event"):
+            for node_id, data in bulk_data.items():
+                if node_id in node_id_to_group_data:
+                    event_id, group_id = (
+                        node_id_to_group_data[node_id][0],
+                        node_id_to_group_data[node_id][1],
+                    )
+                    event = Event(event_id=event_id, project_id=project_id, group_id=group_id)
+                    event.data = data
+                    if (
+                        event
+                        and event.data
+                        and event.data.get("exception")
+                        and hashes.get(group_id)
+                    ):
+                        grouping_info = get_grouping_info(
+                            None, project=project, event_id=event.event_id, event=event
+                        )
+                        stacktrace_string = get_stacktrace_string(grouping_info)
+                        if stacktrace_string == "":
+                            invalid_event_ids.add(event_id)
+                            continue
+                        group_data.append(
+                            CreateGroupingRecordData(
+                                project_id=project_id,
+                                message=messages[group_id],
+                                hash=hashes[group_id],
+                            )
+                        )
+                        stacktrace_strings.append(stacktrace_string)
+                        bulk_event_ids.add(event_id)
+                    else:
+                        invalid_event_ids.add(event_id)
+
+        metrics.gauge(
+            f"{BACKFILL_NAME}._lookup_event_bulk.hit_ratio",
+            round(len(bulk_data.items()) / len(rows)) * 100,
+            sample_rate=1.0,
+        )
+
+        return (
+            bulk_event_ids,
+            invalid_event_ids,
+            GroupStacktraceData(data=group_data, stacktrace_list=stacktrace_strings),
+        )
+
+
+# add timing call for total time of task, nodestore, snuba, seer
+def lookup_group_data_stacktrace_single(
+    project: Project, event_id: str, group_id: int, message: str, hash: str
+) -> tuple[CreateGroupingRecordData | None, str]:
+    with metrics.timer(f"{BACKFILL_NAME}.lookup_event_single", sample_rate=1.0):
```

Comment:
can use @metrics.wraps on the function here instead of the timer context to avoid the indentation